src/app/transactions.component.spec.ts

```diff
@@ -0,0 +1,61 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import TransactionsComponent from './transactions.component';
+
+describe('TransactionsComponent', () => {
+    let fixture: ComponentFixture<TransactionsComponent>;
+    let component: TransactionsComponent;
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [TransactionsComponent],
+    });
+    fixture = TestBed.createComponent(TransactionsComponent);
+    component = fixture.componentInstance;
+  });
+
+//   it('form submitted', () => {
+//     const spy = spyOn(component, 'addTransaction').and.callThrough();
+//     fixture.debugElement.query(By.css('#addTransaction')).triggerEventHandler('click', null);
+//     expect(spy).toHaveBeenCalled();
+//   })
+
+    it('transaction added successfully', () => {
+        const amountOfTransactions = component.transactions().length;
+        component.addTransaction({
+            amount: 15,
+            title: 'Some Title',
+        });
+        fixture.detectChanges();
+        const amountAfterAdd = component.transactions().length;
+        expect(amountAfterAdd).toBe(amountOfTransactions + 1);
+    });
+
+    it('transaction deleted successfully', () => {
+        const [firstTransaction] = component.transactions();
+        const amountOfTransactions = component.transactions().length;
+        component.deleteTransaction(firstTransaction.id);
+        fixture.detectChanges();
+        const amountAfterDelete = component.transactions().length;
+        const amountAfterDeleteShouldBe = amountOfTransactions - 1
+        expect(amountAfterDelete).toBe(amountAfterDeleteShouldBe);
+        const transactionRows = fixture.debugElement.queryAll(By.css('.transaction-tr'));
+        expect(transactionRows.length).toBe(amountAfterDeleteShouldBe);
+    });
+
+    it('dialog opened successfully', () => {
+        component.openAddTransactionDialog();
+        fixture.detectChanges();
+        expect(component.addTransactionDialogOpen()).toBe(true);
+    });
+
+    it('check balance calculation after add', () => {
+        const currentBalance = component.currentBalance();
+        component.addTransaction({
+            amount: 15,
+            title: 'Some Title',
+        });
+        fixture.detectChanges();
+        const balanceAfterAdd = component.currentBalance();
+        expect(balanceAfterAdd).toEqual(currentBalance+15);
```

Comment:
Suddenly you remember you can just add numbers with no intermediate variable here 😁